### PR TITLE
{vis} [{intel,foss}-2015a] matplotlib-1.4.3-Python-2.7.9 (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

File: easybuild/easyconfigs/m/mock/mock-1.0.1-foss-2015a-Python-2.7.9.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+python = "Python"
+pythonversion = '2.7.9'
+#mock is part of Python 3.3
+pyshortver = ".".join(pythonversion.split(".")[:-1])
+
+versionsuffix = "-%s-%s" % (python, pythonversion)
+
+dependencies = [
+    (python, pythonversion),
```
Comment (boegel) [inline easybuild/easyconfigs/m/mock/mock-1.0.1-foss-2015a-Python-2.7.9.eb]: this has no other dependencies except for Python, so why not include it in the Python easyconfig as an extension?

any particular reason?

File: easybuild/easyconfigs/p/pytz/pytz-2014.10-foss-2015a-Python-2.7.9.eb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+sources = [SOURCE_TAR_GZ]
+
+python = "Python"
+pythonversion = '2.7.9'
+pyshortver = ".".join(pythonversion.split(".")[:-1])
+
+versionsuffix = "-%s-%s" % (python, pythonversion)
+
+dependencies = [
+    (python, pythonversion),
```
Comment (boegel) [inline easybuild/easyconfigs/p/pytz/pytz-2014.10-foss-2015a-Python-2.7.9.eb]: same here, include as an extension in the Python easyconfig instead?

File: easybuild/easyconfigs/m/matplotlib/Qhull-intel-fix.patch [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# they use some weird template thing in pure C code to avoid
```
Comment (boegel) [inline easybuild/easyconfigs/m/matplotlib/Qhull-intel-fix.patch]: shouldn't we version this patch file, i.e. `matplotlib-1.4.3_Qhull-intel-fix.patch`?

File: easybuild/easyconfigs/m/matplotlib/matplotlib-1.4.3-intel-2015a-Python-2.7.9.eb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+description = """matplotlib is a python 2D plotting library which produces publication quality figures in a variety of
+ hardcopy formats and interactive environments across platforms. matplotlib can be used in python scripts, the python
+ and ipython shell, web application servers, and six graphical user interface toolkits."""
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+
+patches = [('Qhull-intel-fix.patch')]
```
Comment (boegel) [inline easybuild/easyconfigs/m/matplotlib/matplotlib-1.4.3-intel-2015a-Python-2.7.9.eb]: version this? `patches = [('matplotlib-%(version)s_Qhull-intel-fix.patch')]`

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by hajgato** [comment]
@boegel Requested changes are done. Now needs https://github.com/hpcugent/easybuild-easyconfigs/pull/1462

File: easybuild/easyconfigs/m/matplotlib/matplotlib-1.4.3-intel-2015a-Python-2.7.9.eb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+python = "Python"
+pythonversion = '2.7.9'
+pyshortver = ".".join(pythonversion.split(".")[:-1])
+
+versionsuffix = "-%s-%s" % (python, pythonversion)
+
+dependencies = [
+    (python, pythonversion),
+    ('zlib', '1.2.8'),
```
Comment (boegel) [inline easybuild/easyconfigs/m/matplotlib/matplotlib-1.4.3-intel-2015a-Python-2.7.9.eb]: is the explicit zlib dep needed? the Python dep will already pull in zlib

**Comment by hajgato** [comment]
@boegel zlib dep removed

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e1240f21d687c24c89a8 for a full test report.

**Comment by boegel** [comment]
Looks great, thanks @hajgato!

**Pull request merged**